README.md

```diff
@@ -0,0 +1,92 @@
+Keep multiline test case inputs and outputs out of your Dart test code, and
+write them directly as text files.
+
+## Example
+
+Take the following test case for String's `toUpperCase()`:
+
+```dart
+void main() {
+  test('Works on multiline strings', () {
+    var input = r'''This is the first line.
+This is another line.
+
+This is another paragraph.''';
+
+    var expectedOutput = r'''THIS IS THE FIRST LINE.
+THIS IS ANOTHER LINE.
+
+THIS IS ANOTHER PARAGRAPH.''';
+
+    expect(input.toUpperCase(), equals(expectedOutput));
+  });
+}
+```
+
+Multiline strings break the visual flow of code indentations. Additionally, when
+newline characters are important, you may not be able to just write a newline
+after `r'''`, or before the closing `''';`, which makes it hard to scan for
+where multiline Strings begin and end. Instead, let's add write this test case,
+and a few more, in a separate text file:
+
+```none
+>>> Works on simple strings
+This is a single line.
+<<<
+THIS IS A SINGLE LINE.
+>>> Does nothing to upper case strings
+THIS IS ALREADY UPPER CASE.
+<<<
+THIS IS ALREADY UPPER CASE.
+>>> Works on multiline strings
+This is the first line.
+This is another line.
+
+This is another paragraph.
+<<<
+THIS IS THE FIRST LINE.
+THIS IS ANOTHER LINE.
+
+THIS IS ANOTHER PARAGRAPH.
+```
+
+We can quickly create tests over these data cases with some Dart:
+
+```dart
+library touppercase.tests;
+
+void main() {
+  for (var dataCase in dataCasesUnder(library: #touppercase.tests)) {
```

Comment:
There is no other way to get a directory relative to the executing code. (Unless there recently is.) The mirrors incantation comes from https://github.com/dart-lang/test/issues/110.